sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2579,8 +2579,15 @@ class Dataset[T] private[sql](
    * @group action
    * @since 1.6.0
    */
-  def foreach(f: T => Unit): Unit = withNewExecutionId {
```

Comment:
how about we add a `withNewRDDExecutionId`?

Reply:
Ok.